src/main/java/io/crate/jmx/recorder/Recorder.java

```diff
@@ -45,6 +45,14 @@ default boolean recordBean(String domain,
                                                 " cannot be called with CompositeData bean value");
     }
 
+    default boolean recordBean(String domain,
```

Comment:
I wonder if it would make sense to simplify the `Recorder` Interface to a single method of the signature:

```
    default boolean recordBean(String domain,
                               String attrName,
                               Object beanValue,
                               MetricSampleConsumer metricSampleConsumer);
```

Then handle the type specific logic inside the `Recorder` implementations and also remove all the type specific handling inside `CrateCollector`.

Reply:
Not sure how that would look like, I guess it's worth a shot as a follow up? :+1: